packages/@react-aria/tag/src/useTag.ts

```diff
@@ -10,42 +10,42 @@
  * governing permissions and limitations under the License.
  */
 
-import {ButtonHTMLAttributes, HTMLAttributes, KeyboardEvent, ReactNode} from 'react';
-import {DOMProps, Removable} from '@react-types/shared';
+import {ButtonHTMLAttributes, HTMLAttributes, KeyboardEvent, useRef} from 'react';
 import {filterDOMProps, mergeProps, useId} from '@react-aria/utils';
 // @ts-ignore
 import intlMessages from '../intl/*.json';
+import {useGridCell, useGridRow} from '@react-aria/grid';
 import {useMessageFormatter} from '@react-aria/i18n';
 
 
-export interface AriaTagProps extends Removable<ReactNode, void>, DOMProps {
-  children?: ReactNode,
-  isDisabled?: boolean,
-  validationState?: 'invalid' | 'valid',
-  isSelected?: boolean,
-  role?: string
-}
-
 export interface TagAria {
   tagProps: HTMLAttributes<HTMLElement>,
   labelProps: HTMLAttributes<HTMLElement>,
   clearButtonProps: ButtonHTMLAttributes<HTMLButtonElement>
 }
 
-export function useTag(props: AriaTagProps): TagAria {
+export function useTag(props, state): TagAria {
   const {
     isDisabled,
-    validationState,
     isRemovable,
-    isSelected,
     onRemove,
     children,
-    role
+    item
   } = props;
   const formatMessage = useMessageFormatter(intlMessages);
   const removeString = formatMessage('remove');
   const tagId = useId();
   const buttonId = useId();
+  let tagRef = useRef();
+  let labelRef = useRef();
+
+  let {rowProps} = useGridRow({
+    node: item
+  }, state, tagRef);
+  let {gridCellProps} = useGridCell({
+    node: item.childNodes[0],
+    focusMode: 'cell'
+  }, state, labelRef);
 
   function onKeyDown(e: KeyboardEvent<HTMLElement>) {
     if (e.key === 'Delete' || e.key === 'Backspace') {
```

Comment:
Should hitting Space when the Tag is focused fire onRemove? It did that for v2 Tags, unsure if that should be carried over

Reply:
I don't have preference either way. Might be a good idea to ask someone who would know more